tracks/python/exercises/armstrong-numbers/mentoring.md

```diff
@@ -0,0 +1,60 @@
+# Mentoring
+
+## Problem and challenges
+
+The problem requires the student to sum each digit of an integer or _natural number_ raised to the number of digits' power.
+
+## Reasonable solutions
+
+This exercise can be solved fairly straight forward:
+
+```python
+def is_armstrong(number):
+    digits = (int(d) for d in str(number))
+    count = len(str(number))
+    total = sum(digit ** count for digit in digits)
+    return number == total
+```
+
+This may also be written without the generator.
+This avoids the more complex generator type at the expense of a more complex sum line.
+
+```python
+def is_armstrong(number):
+    count = len(str(number))
+    total = sum(int(digit) ** count for digit in str(number))
+    return number == total
+```
+
+### Common Suggestions
+
+#### Generator Expressions
+
+```python
+    total = sum([digit ** count for digit in digits])
+```
```

Comment:
```suggestion
```python
    # With an intermediate list passed to sum.
    total = sum([digit ** count for digit in digits])
    
    # With a generator expression passed to sum
    total = sum(digit ** count for digit in digits)
```
```